pkg/services/ngalert/state/template/funcs.go

```diff
@@ -89,3 +91,47 @@ func tableLinkFunc(data string) string {
 	expr := url.QueryEscape(q.Expr)
 	return fmt.Sprintf(`/explore?left={"datasource":%[1]q,"queries":[{"datasource":%[1]q,"expr":%q,"instant":true,"range":false,"refId":"A"}],"range":{"from":"now-1h","to":"now"}}`, datasource, expr)
 }
+
+func contains[T comparable](slice []T, elem T) bool {
+	for _, e := range slice {
+		if e == elem {
+			return true
+		}
+	}
+	return false
+}
+
+func deduplicateLabelsFunc(values map[string]Value) Labels {
+	type uniqueLabelVals map[string]struct{}
+
+	labels := make(map[string]uniqueLabelVals)
+	for _, value := range values {
+		for k, v := range value.Labels {
+			if _, ok := labels[k]; !ok {
+				labels[k] = uniqueLabelVals{v: struct{}{}}
+				continue
+			}
+			if _, ok := labels[k][v]; !ok {
+				labels[k][v] = struct{}{}
+			}
```

Comment:
wdyt about this? I think it's a little bit clearer
```suggestion
            var ul uniqueLabelVals
            var ok bool
			if ul, ok = labels[k]; !ok {
			    ul = uniqueLabelVals{}
				labels[k] = ul
			}
			ul[v] = struct{}{}
```